tests/e2e/lifecycle/mcg/test_mcg_namespace_lifecyle_crd.py

```diff
@@ -266,3 +269,181 @@ def test_mcg_namespace_lifecycle_crd(
+            interface=obc_interface,
+            bucketclass=bucketclass_dict,
+        )[0]
+        logger.info(f"Cache bucket: {ns_bucket.name} created")
+        target_bucket = ns_bucket.bucketclass.namespacestores[0].uls_name
+
+        # Write to cache
+        logger.info(f"Writing object on cache bucket: {ns_bucket.name}")
+        assert s3_put_object(
+            mcg_obj, ns_bucket.name, object_key, data
+        ), "Failed: PutObject"
+        wait_for_cache(mcg_obj, ns_bucket.name, [object_key])
+
+        # Write to hub and read from cache
+        logger.info("Setting up test files for upload")
+        setup_base_objects(awscli_pod, amount=3)
+        logger.info(f"Uploading objects to ns target: {target_bucket}")
+        sync_object_directory(
+            awscli_pod,
+            src=MCG_NS_ORIGINAL_DIR,
+            target=f"s3://{target_bucket}",
+            signed_request_creds=s3_creds,
+        )
+        sync_object_directory(
+            awscli_pod, f"s3://{ns_bucket.name}", MCG_NS_RESULT_DIR, mcg_obj
+        )
+
+        # Read cached object
+        assert s3_get_object(mcg_obj, ns_bucket.name, object_key), "Failed: GetObject"
+
+        # Read stale object(ttl expired)
+        sleep(bucketclass_dict["namespace_policy_dict"]["ttl"] / 1000)
+        logger.info(f"Get object on cache bucket: {ns_bucket.name}")
+        assert s3_get_object(mcg_obj, ns_bucket.name, object_key), "Failed: GetObject"
+
+        # List on cache bucket
+        list_response = s3_list_objects_v1(s3_obj=mcg_obj, bucketname=ns_bucket.name)
+        logger.info(f"Listed objects: {list_response}")
+
+        # Delete object from cache bucket
+        s3_delete_object(mcg_obj, ns_bucket.name, object_key)
+        sleep(5)
+        # Try to read deleted object
+        try:
+            s3_get_object(mcg_obj, ns_bucket.name, object_key)
+        except boto3exception.ClientError:
+            logger.info("object deleted successfully")
+
+        # Validate deletion on the hub
+        if (
+            constants.RGW_PLATFORM
+            in bucketclass_dict["namespace_policy_dict"]["namespacestore_dict"]
+        ):
+            obj_list = list(
+                cld_mgr.rgw_client.client.Bucket(target_bucket).objects.all()
+            )
+        else:
+            obj_list = list(
+                cld_mgr.aws_client.client.Bucket(target_bucket).objects.all()
+            )
+        if object_key in obj_list:
+            raise UnexpectedBehaviour("Object was not deleted from cache properly")
+
+        # Recreate and validate object
+        assert s3_put_object(
+            mcg_obj, ns_bucket.name, object_key, data
+        ), "Failed: PutObject"
+        assert s3_get_object(mcg_obj, ns_bucket.name, object_key), "Failed: GetObject"
+
+        logger.info(f"Deleting cache bucket {ns_bucket.name}")
+        curr_ns_store = ns_bucket.bucketclass.namespacestores[0]
+        ns_bucket.delete()
+        new_bucket_class = {
```

Comment:
you can also just edit the `bucketclass_dict` param instead of creating a new dict